src/algebra/group_power/order.lean

```diff
@@ -356,6 +356,22 @@ end
 theorem abs_le_of_sq_le_sq' (h : x^2 ≤ y^2) (hy : 0 ≤ y) : -y ≤ x ∧ x ≤ y :=
 abs_le.mp $ abs_le_of_sq_le_sq h hy
 
+lemma sq_eq_sq_iff_abs_eq_abs (x y : R) : x^2 = y^2 ↔ |x| = |y| :=
+⟨λ h, (abs_le_abs_of_sq_le_sq h.le).antisymm (abs_le_abs_of_sq_le_sq h.ge),
+ λ h, by rw [←sq_abs, h, sq_abs]⟩
+
+@[simp] lemma sq_eq_one_iff (x : R) : x^2 = 1 ↔ x = 1 ∨ x = -1 :=
+by rw [←abs_eq_abs, ←sq_eq_sq_iff_abs_eq_abs, one_pow]
+
+lemma sq_ne_one_iff (x : R) : x^2 ≠ 1 ↔ x ≠ 1 ∧ x ≠ -1 :=
+(not_iff_not.2 (sq_eq_one_iff _)).trans not_or_distrib
+
+@[simp] lemma sq_le_one_iff_abs_le (x : R) : x^2 ≤ 1 ↔ |x| ≤ 1 :=
```

Comment:
```suggestion
@[simp] lemma sq_le_one_iff_abs_le_one (x : R) : x^2 ≤ 1 ↔ |x| ≤ 1 :=
```

Reply:
Do you want this change for `sq_lt_one_iff_abs_lt` as well? And what about for `sq_eq_one_iff`? Also, isn't the additional `_one` redundant since it's implied by the earlier parts of the name.

Reply:
For `sq_lt_one_iff_abs_lt` too. For `sq_eq_one_iff` no, because you are already not including the RHS. It is somewhat redundant, but other lemmas have been following this and I suspect it'll help finding them.

Reply:
Fair point, changed!